modules/file_writer_module/src/parquet_fb_impl.cpp

```diff
@@ -0,0 +1,368 @@
+        auto inputData = static_cast<InputType*>(packet.getData());
+
+        int64_t* domainData = static_cast<int64_t*>(packet.getDomainPacket().getData());
+        
+        using BuilderType = typename SampleToBuilderType<InputSampleType>::Type;
+        auto& dataTable = dataTablesMap.at(domainGlobalId);
+        auto builder = static_cast<BuilderType*>(dataTable.dataBuilderMap[globalId].get());
+
+        if (downScalingDevider > 1)
+        {
+            adaptedSampleCount = 0;
+            for (size_t i = 0; i < sampleCount; ++i)
+            {
+                dataTable.dividerCounter++;
+                if (dataTable.dividerCounter == downScalingDevider)
+                {
+                    inputData[adaptedSampleCount] = inputData[i];
+                    domainData[adaptedSampleCount] = domainData[i];
+                    adaptedSampleCount++;
+                    dataTable.dividerCounter = 0;
+                }
+            }
+        }
+
+        if (adaptedSampleCount == 0)
+        {
+            // Nothing to store
+            return;
+        }
+
+        PARQUET_THROW_NOT_OK(builder->AppendValues(inputData, adaptedSampleCount));
+        // Domain data only add if it is the first domain package.
+        if (dataTable.lastDomainValue  != domainData[adaptedSampleCount-1])
+        {
+            PARQUET_THROW_NOT_OK(dataTable.domainBuilder.AppendValues(domainData, adaptedSampleCount));
+            dataTable.lastDomainValue = domainData[adaptedSampleCount-1];
+        }
+        dataTable.empty = false;
+
+        if (currentTime > dataTable.batchCylceReached)
+        {
+            std::shared_ptr<arrow::Table> table = generateTable(dataTable);
+            writeParquetFile(*table, dataTable.tableNr, dataTable.batchCount);
+            ++dataTable.batchCount;
+            dataTable.batchCylceReached = currentTime + batchCycle;
+        }
+    }
+    else
+    {
+        LOG_T("Data tables have not the domain id: {} stored.", inputPort.getGlobalId());
+    }
+}
+
+void ParquetFbImpl::onConnected(const InputPortPtr& inputPort)
+{
+    std::scoped_lock lock(sync);
+    time_t currentTime;
+    std::string domainId = inputPort.getSignal().getDomainSignal().getGlobalId();
+    std::string signalId = inputPort.getSignal().getGlobalId();
+    auto sampleType = inputPort.getSignal().getDescriptor().getSampleType();
+
+    inputPortSignalContextMap.emplace(std::piecewise_construct, std::forward_as_tuple(inputPort.getGlobalId()), 
+                                                        std::forward_as_tuple(signalId, domainId, sampleType));
+
+
+    if (dataTablesMap.find(domainId) == dataTablesMap.end()) 
+    {
+        tableNumberCount++;
+        dataTablesMap.emplace(std::piecewise_construct, std::forward_as_tuple(domainId), 
+                                                        std::forward_as_tuple(domainId, tableNumberCount));
+    }
+
+    switch(sampleType)
```

Comment:
@JakaMohorkoDS : With the change and other it is possible to support all Scalar Types.